src/dict.js

```diff
@@ -157,47 +158,88 @@ Sk.builtin.dict.prototype.sq$contains = function (ob) {
 };
 
 Sk.builtin.dict.prototype.mp$ass_subscript = function (key, w) {
-    var k = kf(key);
-    var bucket = this.buckets[k.v];
-    var item;
-
-    if (bucket === undefined) {
-        // New bucket
-        bucket = {$hash: k, items: [
-            {lhs: key, rhs: w}
-        ]};
-        this.buckets[k.v] = bucket;
-        this.size += 1;
+    const hash = kf(key);
+    if (hash[0] === "_") {
+        // we have a string so pass it to the dictionary 
+        if (this.entries[hash] === undefined) {
+            this.size += 1;
+            this.entries[hash] = {lhs: key};
```

Comment:
you should pre-initialize `rhs` to be `w` here too, otherwise the rhs property will go out-of-object (on V8 at least).